model-optimizer/extensions/front/FakeOutputResolver.py

```diff
@@ -0,0 +1,53 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.ops.elementwise import Add
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Node, Graph, rename_nodes, rename_node
+
+
+class FakeOutputResolver(FrontReplacementSubgraph):
+    enabled = True
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('fake_output', {'op': 'FakeOutput'})
+            ],
+            edges=[],
+        )
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        fake_output = match['fake_output']
+        name = fake_output.soft_get('name', fake_output.id)
+
+        fake_output.in_port(0)
+        producer = fake_output.in_port(0).get_source().node
+        producer_outputs = 0
+        for port in producer.out_ports().values():
+            if not port.disconnected():
+                producer_outputs +=1
+        if producer_outputs != 1:
+            add = create_op_with_const_inputs(graph, Add, {1: int64_array(0)}, {'can_be_fused': False})
```

Comment:
Please, leave a comment that this code relies on MO transformation which updates the Const type for elementwise operations in case of input data types mismatch